modules/unsupported/wfs-ng/src/test/java/org/geotools/data/wfs/internal/AbstractWFSStrategyTest.java

```diff
@@ -68,4 +72,123 @@ public void testSplitFilters() throws IOException, ServiceException {
         Assert.assertEquals(2, filters.length);
         Assert.assertTrue(filters[0].evaluate(filter));
     }
+
+    /**
+     * Test method for {@link
+     * org.geotools.data.wfs.internal.AbstractWFSStrategy#buildUrlGET(WFSRequest)}, to check if it
+     * produces the <code>STARTINDEX</code> query parameter also for a WFS 1.X request.
+     */
+    @Test
+    public void testWfs1_XUrlBuildingWithStartIndexParameter() throws Exception {
+
+        WFSStrategy strategy = new StrictWFS_1_x_Strategy();
+
+        HTTPResponse httpResponse =
+                new TestHttpResponse(
+                        "text/xml", "UTF-8", url("GeoServer_1.7.x/1.1.0/GetCapabilities.xml"));
+        WFSGetCapabilities capabilities =
+                new GetCapabilitiesResponse(httpResponse, null).getCapabilities();
+        strategy.setCapabilities(capabilities);
+
+        GetFeatureRequest request = new GetFeatureRequest(new WFSConfig(), strategy);
+
+        request.setTypeName(new QName("example"));
+        request.setFilter(Filter.INCLUDE);
+        request.setStartIndex(100);
+
+        URL url = strategy.buildUrlGET(request);
+
+        Assert.assertTrue(url.getQuery().contains("STARTINDEX=100"));
```

Comment:
The 1.x strategy declares that it cannot do offset. If we send the startIndex anyways in the URL, the server honors it, and then the store also applies a local offset assuming it was not sent, the result will not be the expected one... or not?

Reply:
Better enforce the absence of the parameter, then. Thanks.